plugins/vendor/webex/services/WebexAPIService.php

```diff
@@ -0,0 +1,327 @@
+<?php
+/**
+ * @service webexAPI
+ * @package plugins.WebexAPIDropFolder
+ * @subpackage api.services
+ */
+class WebexAPIService extends KalturaBaseService
+{
+	const AUTH_CODE = 'code';
+	const STATE = 'state';
+	const INTEGRATION_CODE = 'integrationCode';
+	const REGISTRATION_PAGE_PATH = '/../lib/api/webPage/KalturaWebexAPIRegistrationPage.html';
+	const LOGIN_PAGE_PATH = '/../lib/api/webPage/KalturaWebexAPILoginPage.html';
+	const REGIONAL_REDIRECT_PAGE_PATH = '/../lib/api/webPage/KalturaWebexAPIRegionalRedirectPage.html';
+	
+	protected static $PARTNER_NOT_REQUIRED_ACTIONS = array();
+	
+	protected static $webexIntegration;
+	
+	/**
+	 * no partner will be provided by vendors as this called externally and not from kaltura
+	 * @param string $actionName
+	 * @return bool
+	 */
+	protected function partnerRequired($actionName)
+	{
+		return in_array($actionName, self::$PARTNER_NOT_REQUIRED_ACTIONS);
+	}
+	
+	/**
+	 * @param $serviceId
+	 * @param $serviceName
+	 * @param $actionName
+	 * @throws KalturaAPIException
+	 */
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+	}
+	
+	/**
+	 * @param $accountId
+	 * @param bool $includeDeleted
+	 * @return null|WebexAPIVendorIntegration
+	 * @throws PropelException
+	 */
+	protected static function getWebexAPIIntegrationByAccountId($accountId, $includeDeleted = false)
+	{
+		if ($includeDeleted)
+		{
+			self::$webexIntegration = VendorIntegrationPeer::retrieveSingleVendorPerPartnerNoFilter($accountId, VendorTypeEnum::WEBEX_API_ACCOUNT);
+		}
+		else
+		{
+			self::$webexIntegration = VendorIntegrationPeer::retrieveSingleVendorPerPartner($accountId, VendorTypeEnum::WEBEX_API_ACCOUNT);
+		}
+		
+		return self::$webexIntegration;
+	}
+	
+	/**
+	 * @param WebexAPIVendorIntegration $webexIntegration
+	 * @param $accountId
+	 * @param $ks
+	 * @throws Exception
+	 */
+	protected static function loadSubmitPage(WebexAPIVendorIntegration $webexIntegration, $accountId, $ks)
```

Comment:
looks same as zoom so why not create a function that gets the registration path as param